lib/http.c

```diff
@@ -517,14 +521,14 @@ CURLcode Curl_http_auth_act(struct connectdata *conn)
   if(conn->bits.user_passwd &&
      ((data->req.httpcode == 401) ||
       (conn->bits.authneg && data->req.httpcode < 300))) {
-    pickhost = pickoneauth(&data->state.authhost);
+    pickhost = pickoneauth(&data->state.authhost, authmask);
     if(!pickhost)
       data->state.authproblem = TRUE;
   }
   if(conn->bits.proxy_user_passwd &&
      ((data->req.httpcode == 407) ||
       (conn->bits.authneg && data->req.httpcode < 300))) {
-    pickproxy = pickoneauth(&data->state.authproxy);
+    pickproxy = pickoneauth(&data->state.authproxy, authmask);
```

Comment:
The `oauth_bearer` is however only documented and implemented to work for host auth... Which really begs the question what libcurl should do if this is asked to be used for proxy auth. Maybe this should just *always* disable `CURLAUTH_BEARER` for proxy since there's no option to set the bearer string for it?

Reply:
Right, and this code is guarded behind a test for `proxy_user_password`, too...